platforms/quorum/configuration/roles/setup/constellation-node/tasks/main.yaml

```diff
@@ -0,0 +1,45 @@
+---
+- name: Register temporary directory
+  tempfile: 
+    state: directory
+  register: tmp_directory
+
+- name: check constellation
+  stat:
+    path: "{{ bin_install_dir }}/constellation-node"
+  register: constellation_stat_result
+
+- name: Finding the release for os
+  shell: |
+    lsb_release -r
+  register: release
+
+- name: Release version
+  set_fact: release_version={{ release.stdout | regex_replace('Release:\t') | regex_replace('\.') }}
+
+- name: Download the constellation-node binary
+  get_url: 
+    url: "{{ constellation.repo }}"
+    dest: "{{ tmp_directory.path }}"
+  when: constellation_stat_result.stat.exists == False
+
+- name: unzip the tar file
+  shell: |
```

Comment:
Check if https://docs.ansible.com/ansible/latest/modules/unarchive_module.html can be used instead of shell